microprofile.ls/org.eclipse.lsp4mp.ls/src/test/java/org/eclipse/lsp4mp/services/ApplicationPropertyExpressionCompletionTest.java

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4mp.services;
+
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.c;
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.r;
+import static org.eclipse.lsp4mp.services.MicroProfileAssert.testCompletionFor;
+
+import org.eclipse.lsp4mp.ls.commons.BadLocationException;
+import org.junit.Test;
+
+public class ApplicationPropertyExpressionCompletionTest {
+
+	// Tests for where property expression completion should open
+
+	@Test
+	public void justDollarSignNoNewline() throws BadLocationException {
+		String text = //
+				"test.property = hello\n" + //
+				"other.test.property = $|";
+		testCompletionFor(text, true, c("${test.property}", "${test.property}", r(1, 22, 1, 23)));
+	}
+
+	@Test
+	public void justDollarSignNewline() throws BadLocationException {
+		String text = //
+				"test.property = hello\n" + //
+				"other.test.property = $|\n";
+		testCompletionFor(text, true, c("${test.property}", "${test.property}", r(1, 22, 1, 23)));
+	}
+
+	@Test
+	public void beforeDollarSign() throws BadLocationException { // FIXME:
+		String text = //
+				"test.property = hello\n" + //
+				"other.test.property = |$\n";
+		testCompletionFor(text, true, c("${test.property}", "${test.property}", r(1, 22, 1, 23)));
+	}
```

Comment:
Should I look into changing this behaviour, or does this behaviour make sense?

Reply:
I think in this case there shouldn't be completion options, right @fbricon, @angelozerr ?

Reply:
I addressed this in 01a207e